google/cloud/pubsublite/cloudpubsub/internal/client_multiplexer.py

```diff
@@ -75,52 +68,45 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
 
 class AsyncClientMultiplexer(Generic[_Key, _Client]):
-    _OpenedClientFactory = Callable[[], Awaitable[_Client]]
+    _OpenedClientFactory = Callable[[_Key], Awaitable[_Client]]
     _ClientCloser = Callable[[_Client], Awaitable[None]]
 
+    _factory: _OpenedClientFactory
     _closer: _ClientCloser
-    _lock: asyncio.Lock
-    _live_clients: Dict[_Key, _Client]
+    _live_clients: Dict[_Key, Awaitable[_Client]]
```

Comment:
There was a lot of time in acquiring the lock in the publish hotpath- this change makes the lock not necessary